server/src/actors/events.rs

```diff
@@ -360,48 +341,62 @@ impl EventProcessor {
+    fn extract_event(
+        &self,
+        envelope: &mut Envelope,
+    ) -> Result<Option<Annotated<Event>>, ProcessingError> {
         // Remove all items first, and then process them. After this function returns, only
         // attachments can remain in the envelope. The event will be added again at the end of
         // `process_event`.
         let event_item = envelope.take_item_by_type(ItemType::Event);
         let security_item = envelope.take_item_by_type(ItemType::SecurityReport);
+
         let form_item = envelope.take_item_by_type(ItemType::FormData);
-        let attachment_item = envelope.take_item_by_name(ITEM_NAME_EVENT);
-        let breadcrumbs_item1 = envelope.take_item_by_name(ITEM_NAME_BREADCRUMBS1);
-        let breadcrumbs_item2 = envelope.take_item_by_name(ITEM_NAME_BREADCRUMBS2);
+        let attachment_item = envelope.take_item_by_attachment_type(AttachmentType::Event);
+        let breadcrumbs_item1 = envelope.take_item_by_attachment_type(AttachmentType::Breadcrumb);
+        let breadcrumbs_item2 = envelope.take_item_by_attachment_type(AttachmentType::Breadcrumb);
 
         if let Some(item) = event_item {
             log::trace!("processing json event {}", envelope.event_id());
-            return metric!(timer("event_processing.deserialize"), {
-                self.event_from_json_payload(item)
-            });
+            return Ok(Some(metric!(timer("event_processing.deserialize"), {
+                self.event_from_json_payload(item)?
+            })));
         }
 
         if let Some(item) = security_item {
             log::trace!("processing security report {}", envelope.event_id());
-            return self.event_from_security_report(item);
+            return Ok(Some(self.event_from_security_report(item)?));
         }
 
         if attachment_item.is_some() || breadcrumbs_item1.is_some() || breadcrumbs_item2.is_some() {
             log::trace!("extracting attached event data {}", envelope.event_id());
-            return Self::event_from_attachments(
+            return Ok(Some(Self::event_from_attachments(
                 &self.config,
                 attachment_item,
                 breadcrumbs_item1,
                 breadcrumbs_item2,
-            );
+            )?));
         }
 
         if let Some(item) = form_item {
             log::trace!("extracting form data {}", envelope.event_id());
             let mut value = SerdeValue::Object(Default::default());
             self.merge_formdata(&mut value, item);
-            return Ok(Annotated::deserialize_with_meta(value).unwrap_or_default());
+            return Ok(Some(
+                Annotated::deserialize_with_meta(value).unwrap_or_default(),
+            ));
         }
 
-        log::trace!("creating empty event {}", envelope.event_id());
-        Ok(Annotated::new(Event::default()))
+        if envelope
+            .get_item_by_attachment_type(AttachmentType::Minidump)
```

Comment:
Can we always return None here and check this in the caller, where we already perform a check for the minidump attachment?

Reply:
We have an if block in `fn process`, which is also where unreal is being checked. You could add an Else there.

Reply:
done